### ISSUE TITLE
Handling < 1 sec

### ISSUE THREAD

**Issue by psatyajeet** [philipperemy/lead-lag#11]
I have a question about the statement "Limitations: Only supports up to the second. Everything labeled in milliseconds is not correctly handled (at the moment)."

Do you mean that it only handles up to the second portion of timestamps and ignores the milliseconds? Or do you also mean that you can't introduce lags less than a second because it would mess up the calculations in some way? 

I'm thinking of trying lags less than 1 second but want to be aware and understand the reasons for the stated limitation

**Comment by philipperemy** [comment]
@psatyajeet the reason why it's limited up to the second for now is because I have an interval [-max_lag, max_lag] sampled by increments of 1 second.
At the moment, if we introduce lags less than a second, it would mess up the calculations in some way.
It's definitely possible to have less than 1 second. The code just needs to be updated :)

**Comment by psatyajeet** [comment]
That makes sense! Do you have any suggestions on what code would need to be updated? I’m particularly interested in testing lags between 0-1 seconds so anything you can point me to would be helpful so that I may give it a shot. 

As always, thankful for your great work here — really interesting!!

**Comment by philipperemy** [comment]
Thanks a lot! Much appreciated!

Actually, I think it can technically handle lags between 0-1 seconds with the current code. It's just that we need to give more precise timestamps. For example, if you give 1575001711, it corresponds to a UTC timestamp. But if you give 15750017110 (one additional zero), then you have a 10x resolution. If the max lag range was [-50 seconds, 50 seconds], it would become [-5 seconds, 5 seconds].

The function that handles the timestamp is this one: https://github.com/philipperemy/lead-lag/blob/6988019f5d6f3d8e1b0e7aa56bcffdbb018c1eb0/lead_lag/lead_lag.py#L94

So you might try this approach by basing yourself on this example: https://github.com/philipperemy/lead-lag/blob/master/notebooks/lead_lag_example_2.ipynb

Let me know. I'm very interesting in knowing if it's possible with the current code :)

**Comment by psatyajeet** [comment]
This makes sense. 

I changed how to read timestamps from the csv file: https://github.com/psatyajeet/lead-lag/pull/2/files

Instead of truncating the timestamp at the full seconds, this truncates it after the hundredth of seconds. Then, when you introduce lags between [-1000 centiseconds, 1000 centiseconds], that translates to [-10 seconds, 10 seconds] as you suggested. 

The results look odd with the data set you used in `lead_lag_example_2.ipynb` because that data only has fidelity to the nearest second. I downloaded a day of Coinbase/Kraken data on my own that includes milliseconds and it looks pretty interesting:

![image](https://user-images.githubusercontent.com/1389682/70267663-bdaa4b00-1753-11ea-9783-b1161d94bb96.png)

^That's for ts1 = Coinbase vs. ts2 = Kraken on 2019-08-01 with an estimated lag of 0.7 seconds (70 centiseconds)
